### PR TITLE
JDK-8328000 : Convert /java/awt/im/8154816/bug8154816.java applet test to main

### PR DESCRIPTION
**Pull request by honkar-jdk**
The following InputMethod test, bug8154816.java is converted to main using PassFailJFrame.

- test renamed to PinyinIMCapsTest
- moved one directory above - /java/awt/im/
- html file deleted.

PS: I tried to automate this test using Robot, but was unable to produce the same effect as a keyboard keypress for Pinyin, hence keep it as a manual test.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328000](https://bugs.openjdk.org/browse/JDK-8328000): Convert /java/awt/im/8154816/bug8154816.java applet test to main (**Bug** - P4)

### Reviewers
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer)
 * [Alexander Zvegintsev](https://openjdk.org/census#azvegint) (@azvegint - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18245/head:pull/18245` \
`$ git checkout pull/18245`

Update a local copy of the PR: \
`$ git checkout pull/18245` \
`$ git pull https://git.openjdk.org/jdk.git pull/18245/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18245`

View PR using the GUI difftool: \
`$ git pr show -t 18245`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18245.diff">https://git.openjdk.org/jdk/pull/18245.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18245#issuecomment-1992291943)

### CODE REVIEW

File: test/jdk/java/awt/im/PinyinIMCapsTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ * @summary Caps Lock doesn't work as expected when using Pinyin Simplified input method
+ * @requires (os.family == "mac")
+ * @modules java.desktop/sun.awt
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jtreg.SkippedException Util
+ * @run main/manual PinyinIMCapsTest
+ */
+
+public class PinyinIMCapsTest {
+    private static final String INSTRUCTIONS = """
```
Comment (TejeshR13) [inline test/jdk/java/awt/im/PinyinIMCapsTest.java:43]: Can move `INSTRUCTIONS ` to main.

File: test/jdk/java/awt/im/PinyinIMCapsTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            This test is for OS X only. It verifies if the Caps Lock key works properly
```
Comment (TejeshR13) [inline test/jdk/java/awt/im/PinyinIMCapsTest.java:44]: I guess its not required to mention regarding OS X requirement in test instructions, anyhow its taken care in jtreg tag and also a check has been made before test initialization.

Comment (honkar-jdk) [reply test/jdk/java/awt/im/PinyinIMCapsTest.java]: Updated

File: test/jdk/java/awt/im/PinyinIMCapsTest.java (hunk 1 above)
Comment (honkar-jdk) [reply test/jdk/java/awt/im/PinyinIMCapsTest.java:43]: Some tests follow this pattern and few others have it within main. I usually define it as static var so retaining it the same here.

**Review by TejeshR13 — APPROVED** [review]

**Review by azvegint — APPROVED** [review]

**Pull request closed without merging**